### ISSUE TITLE
Handle errors before returning the fxa account better.

### ISSUE THREAD

**Issue by b4handjr** [mozilla/pytest-fxa#15]
Recently on [addons-server](https://21144-16416867-gh.circle-artifacts.com/0/home/circleci/addons-server/ui-test.html) we ran into an issue with accounts not being verified, or returning the wrong ```x-verify-code```. It turns out that the setup was failing [here](https://github.com/mozilla/pytest-fxa/blob/master/pytest_fxa/plugin.py#L82), but the account was not being deleted.

Thanks to @shane-tomlinson he discovered that PyFxa returns the _first_ email it finds matching within the ```wait_for_email``` method: https://github.com/mozilla/PyFxA/blob/master/fxa/tests/utils.py#L103

I think we should handle creating an email better, as in, if that ```x-verify-code``` does not match, then we should do something and not just allow the account to still exist. Or maybe some more checks to see what email we get back.

**Comment by davehunt** [comment]
If we're using unique email addresses, shouldn't there only be one verification email anyway? Either way, I'm in favour of making this more robust. We could define a cleanup function in the fixture, which we can then call in the event that the create or verify attempts fail, and then also call it after the yield.

**Comment by shane-tomlinson** [comment]
We have resolved [the original issue](https://github.com/mozilla/fxa/issues/305) by manually deleting the emails for uitest@restmail.net. These the things we noticed while debugging:

1. The test runner always uses the same email address, uitest@restmail.net
2. If the test runner fails creating & verifying an account, the [emails for that address are not deleted](https://github.com/mozilla/pytest-fxa/blob/master/pytest_fxa/plugin.py#L86).
3. PyFxA always [returns the first email that matches the predicate](https://github.com/mozilla/PyFxA/blob/97bfe7ad53650393e1b2dffa6786e0cc2072b090/fxa/tests/utils.py#L95), which in this case the predicate is whether an email has x-verify-code header.

Since the verify-code from the first email was always being returned, and then the auth-server was saying "invalid code", the emails for uitest were never deleted. A quick fix was to manually delete all of the emails for uitest.

We use unique email addresses for each FxA functional test, which has a couple of nice properties: 

* Tests can run in parallel
* We don't worry about cleanup, restmail will jettison emails after a time so it's not a big deal to leave emails on the server.

If continuing with a single email address, it seems like pytest_fxa should ensure the emails are deleted if there is a failure creating/verifying the account.
